objects/src/accounts/mod.rs

```diff
@@ -146,12 +147,15 @@ impl ToAdviceInputs for Account {
     fn to_advice_inputs<T: AdviceInputsBuilder>(&self, target: &mut T) {
         // push core items onto the stack
         target.push_onto_stack(&[*self.id, ZERO, ZERO, self.nonce]);
-        target.push_onto_stack(self.vault.root().as_elements());
+        target.push_onto_stack(self.vault.get_commitment().as_elements());
         target.push_onto_stack(&self.storage.root());
         target.push_onto_stack(self.code.root().as_elements());
 
         // extend the merkle store with the storage items
         target.add_merkle_nodes(self.storage.slots().inner_nodes());
-        target.add_merkle_nodes(self.storage.store().inner_nodes())
+        target.add_merkle_nodes(self.storage.store().inner_nodes());
+
+        // extend the merkle store with account vault data
+        target.add_merkle_nodes(self.vault.asset_tree().inner_nodes())
```

Comment:
@bobbinth can you please advise on how I can populate the `AdviceMap` to support operation of the TSMT please.  We should update `to_advice_inputs(..)` for the `Account` object accordingly.

Reply:
I was thinking for this `upper_leaves()` and `bottom_leaves()` iterators could be used.

- `upper_leaves()` returns a tuple `(node_hash, key, value)`. When putting this into the advice map we'd use `node_hash` as key, and `[key, value]` elements for the values.
- `bottom_leaves()` returns `(node_hash, vec![(key, value])`. When putting this into the advice map we'd use `node_hash` as key and `[key, value]` elements (for all tuples) as values.

I don't think we need to adjust the keys based on which tier they fall into. My current thinking is that this will happen on the MASM side.

Reply:
> * `upper_leaves()` returns a tuple `(node_hash, key, value)`. When putting this into the advice map we'd use `node_hash` as key, and `[key, value]` elements for the values.

This is different from the current implementation right?  For this PR I am population the advice map using `key = node_hash` and `value = [remaining_key, value]`.

Reply:
Correct. Instead of doing `value = [remaining_key, value]` we could just do `value = [key, value]` and then inside MASM adjust accordingly (since we know which tier we are dealing with there).

Reply:
Actually, let's come back to this once I'm done with the Tiered SMT implementation in MASM.